### PR TITLE
[feature]: Add option to use custom https certificates with buildpack

### LINKED ISSUE #2945: [feature]: Add option to use custom https certificates with buildpack
**Describe the solution you'd like**
Add option to add custom https certificates to buildpack serve

**Please let us know what packages this feature is in regards to:**
- [ ] `venia-concept`
- [ ] `venia-ui`
- [x] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by arturtreiberg**
## Description
Added two extra env variable definitions for custom https certificate key and cert paths. Also updated buildpack serve function to add https if custom cert paths are present in the configuration

## Related Issue

Closes #2945

### Verification Steps

1. Add .env variable CUSTOM_HTTPS_KEY with absolute file path to the key file
2. Add .env variable CUSTOM_HTTPS_CERT with absolute file path to the cert file
3. Run buildpack serve .

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by jimbo — COMMENTED** [review]
@arturtreiberg Hello, thanks for your contribution! Seems like a natural addition.

How can we test this? Would like to verify that it works. A couple of unit tests would also be a good idea; let me know if I can help there.

**Comment by arturtreiberg** [comment]
Hey @jimbo, sorry for such a late answer.

So testing wise, just adding these .env variables as described in the verification steps above does the trick. After the variables are added, by running `yarn buildpack serve .` does the trick and the https upward server will be started. Logger message should also say that custom certificates are provided.

Unit test wise, I would have to investigate it a little to see how can this be tested and I'm not sure when I will have the time for that :/ but I'm gonna take a look at what I can do

File: packages/pwa-buildpack/lib/Utilities/serve.js [hunk 1]
```diff
@@ -33,6 +35,24 @@ module.exports = async function serve(dirname) {
...
+        const { key, cert } = customHttpsSettings;
+        if (fs.existsSync(key) && fs.existsSync(cert)) {
+            prettyLogger.info(
+                'Custom key and cert paths provided, creating HTTPS server.'
+            );
+            const ssl = {
+                key: fs.readFileSync(key, 'utf8'),
+                cert: fs.readFileSync(cert, 'utf8')
+            };
+            upwardServerOptions.https = ssl;
```
Comment (sirugh) [inline packages/pwa-buildpack/lib/Utilities/serve.js:48]: If I read this file correctly it looks like it is possible for this setting to get overwritten by the [conditional block](
https://github.com/magento/pwa-studio/blob/c64cb3535b5b7e7adabec608f6c372f4e5d3cac6/packages/pwa-buildpack/lib/Utilities/serve.js#L78-L102) further on in the file.

I might add a condition around this line (currently 93):

```es6
if (!upwardServerOptions.https) {
  upwardServerOptions.https = ssl;
}
```

This way, if someone has specified their own key/cert, we won't overwrite it when configuring the `customOrigin` settings.

**Review by sirugh — CHANGES_REQUESTED** [review]
I think this is really close - just a small addition requested. Also, if you're able to add some tests that would be very helpful.

Thanks for your contribution!

**Comment by arturtreiberg** [comment]
> I think this is really close - just a small addition requested. Also, if you're able to add some tests that would be very helpful.
> 
> Thanks for your contribution!

File: packages/pwa-buildpack/lib/Utilities/serve.js (hunk 1 above)
Comment (arturtreiberg) [reply packages/pwa-buildpack/lib/Utilities/serve.js:56]: Thanks for feedback, added it ;)

**Review by jimbo — APPROVED** [review]
Great work, and thanks for adding tests. This works for me, recommend moving to QA. 👍

**Comment by dpatil-magento** [comment]
QA Approved

**Pull request merged**